src/components/ConversationSettings/MentionsSettings.vue

```diff
@@ -0,0 +1,141 @@
+<!--
+  - SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<template>
+	<div v-if="canModerate">
+		<NcCheckboxRadioSwitch :checked="mentionPermissions === MENTION_PERMISSIONS.EVERYONE"
+			:disabled="isMentionPermissionsLoading"
+			type="switch"
+			@update:checked="toggleMentionPermissions">
+			{{ t('spreed', 'Allow participants to mention @all') }}
+		</NcCheckboxRadioSwitch>
+	</div>
+
+	<div v-else>
+		<h5 class="app-settings-section__subtitle">
+			{{ t('spreed', 'Mention permissions') }}
+		</h5>
+		<p>{{ summaryLabel }}</p>
+	</div>
+</template>
+
+<script>
+import { showError, showSuccess } from '@nextcloud/dialogs'
+import { t } from '@nextcloud/l10n'
+
+import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'
+
+import { CONVERSATION } from '../../constants.js'
+
+export default {
+	name: 'MentionsSettings',
+
+	components: {
+		NcCheckboxRadioSwitch,
+	},
+
+	props: {
+		token: {
+			type: String,
+			default: null,
+		},
+
+		canModerate: {
+			type: Boolean,
+			default: true,
+		},
+	},
+
+	emits: ['input'],
+
+	data() {
+		return {
+			mentionPermissions: null,
+			isMentionPermissionsLoading: false,
+			lastNotification: null,
+			MENTION_PERMISSIONS: CONVERSATION.MENTION_PERMISSIONS,
+		}
+	},
```

Comment:
To separate constant property and reactive mutable state:

```suggestion
	setup() {
		const  = CONVERSATION.MENTION_PERMISSIONS
		return {
			MENTION_PERMISSIONS,
		}
	},

	data() {
		return {
			mentionPermissions: null,
			isMentionPermissionsLoading: false,
			lastNotification: null,
		}
	},
```